axiom/vfields.go

```diff
@@ -0,0 +1,127 @@
+package axiom
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type VirtualField struct {
+	// Dataset is the dataset to which the virtual field belongs.
+	Dataset string `json:"dataset"`
+	// Name is the name of the virtual field.
+	Name string `json:"name"`
+	// Expression defines the virtual field's APL.
+	Expression string `json:"expression"`
+	// Description is an optional description of the virtual field.
+	Description string `json:"description,omitempty"`
+	// Type is the type of the virtual field. E.g. string | number
+	Type string `json:"type,omitempty"`
+	// Unit is the unit for the type of data returned by the virtual field.
+	Unit string `json:"unit,omitempty"`
+}
+
+type VirtualFieldWithID struct {
+	VirtualField
+	// ID is the unique identifier of the virtual field.
+	ID string `json:"id"`
+}
+
+// Axiom API Reference: /v2/vfields
+type VirtualFieldsService service
+
+// List all virtual fields for a given dataset.
+func (s *VirtualFieldsService) List(ctx context.Context, dataset string) ([]*VirtualFieldWithID, error) {
+	ctx, span := s.client.trace(ctx, "VirtualFields.List", trace.WithAttributes(
+		attribute.String("axiom.param.dataset", dataset),
+	))
+	defer span.End()
+
+	params := url.Values{}
+	params.Set("dataset", dataset)
+
+	var res []*VirtualFieldWithID
+	if err := s.client.Call(ctx, http.MethodGet, s.basePath+"?"+params.Encode(), nil, &res); err != nil {
+		return nil, spanError(span, err)
+	}
```

Comment:
Please use `AddURLOptions`: https://github.com/axiomhq/axiom-go/blob/main/axiom/datasets.go#L291-L296.

Your best bet is to use an anonymous struct to make this quick and easy, e.g.:

```go
_ = AddURLOptions(s.basePath, struct {
	Dataset string  `url:"dataset"`
}{
	Dataset: dataset,
})
```